test/test_geocaching.py

```diff
@@ -70,39 +75,49 @@ def test_search_quick(self):
     def test_search_quick_match_load(self):
         """Test if quick search results matches exact cache locations."""
         rect = Rectangle(Point(49.73, 13.38), Point(49.74, 13.39))
-        caches = list(self.g.search_quick(rect, strict=True, zoom=15))
-        for cache in caches:
-            try:
-                cache.load()
-                self.assertIn(cache.location, rect)
-            except PMOnlyException:
-                pass
+        with recorder.use_cassette('geocaching_matchload'):
+            # at commit time, this test is an allowed failure. Once this feature is fixed, the
+            # corresponding cassette will have to be deleted and re-recorded.
+            caches = list(self.g.search_quick(rect, strict=True, zoom=15))
+            for cache in caches:
+                try:
+                    cache.load()
+                    self.assertIn(cache.location, rect)
+                except PMOnlyException:
+                    pass
 
 
 class TestLoginOperations(unittest.TestCase):
-
     def setUp(self):
-        self.g = Geocaching()
+        self.g = Geocaching(_session=session)
 
     def test_request(self):
         with self.subTest("login needed"):
             with self.assertRaises(NotLoggedInException):
                 self.g._request("/")
 
     def test_login(self):
```

Comment:
I don't know if it helps, but feel free to split the subtests to separate tests and somehow use the `setUp()` method to get rid of those `session.cookie.clear()` and logout calls. But maybe this is the best solution, I don't know.

Reply:
I might look into refactoring the setup/teardown of tests after some other issues are fixed. Is this related to your other comment: "What do you think about using single setup (login) cassette for all logins?"?

Reply:
Okay. No, please consider these two comments separately.

Reply:
Fixed in be2b8b4d910a4cd17d0a413517a683f52b94a143 (not pushed yet) — the networked tests all inherit from a new class.